python/Scripts/mxvalidate.py

```diff
@@ -0,0 +1,394 @@
+#!/usr/bin/env python
+'''
+Verify that a specified file is a valid MaterialX document.
+
+Doug Smythe
+'''
+
+import sys, os, string
+import MaterialX as mx
+
+
+def usage():
+    print "mxvalidate.py: verify that a specified file is a valid MaterialX document."
+    print "Usage:  mxvalidate.py [options] <file.mtlx>"
+    print "    -h[elp]         Print usage information"
+    print "    -v[erbose]      Print summary of elements found in the document"
+    print "    -vv[eryverbose] Also print material bindings, nodedef interfaces and nodegraph trees"
+    print "    -i[ncludedefs]  Automatically XInclude stdlib_defs"
+    print "    -r[esolve]      Resolve inheritance and string substitutions"
+
+def main():
+    verbose = 0
+    incldefs = 0
+    resolve = 0
+
+    if len(sys.argv) < 2:
+	usage()
+	sys.exit(0)
+    for arg in sys.argv[1:]:
+	if arg in ['-v', '-verbose']:
+	    verbose = 1
+	elif arg in ['-vv', '-veryverbose']:
+	    verbose = 2
+	elif arg in ['-i', '-includedefs']:
+	    incldefs = 1
+	elif arg in ['-r', '-resolve']:
+	    resolve = 1
+	elif arg in ['-h', '-help', '--help']:
+	    usage()
+	    sys.exit(0)
+	else:
+	    filename = arg
+
+    doc = mx.createDocument()
+    mxversion = mx.getVersionString()
+    (mxmajorv, mxminorv, mxbuildv) = mx.getVersionIntegers()
+
+    # The mx.readFromXmlFile() will fail with
+    #   "MaterialX.ExceptionFileMissing" if the file is not found, or
+    #   "MaterialX.ExceptionParseError" if the document is found but not readable.
+    try:
+	mx.readFromXmlFile(doc, filename)
+    except mx.ExceptionFileMissing as err:
+	#TODO: library doesn't correctly set missing filename if it's an XIncluded file (it's blank)
+	print err
+	return
+    except mx.ExceptionParseError as err:
+	m = err.split('(')[1][:-1]
```

Comment:
There's some potential for fragility here, as the text format of the parse error message isn't guaranteed to remain fixed in future versions of MaterialX (e.g. it may later contain additional information including multiple string literals).  For future robustness, I would recommend just printing the error message, as is done in the ExceptionFileMissing case.

Reply:
done